WordPress/gutenberg
# Block variations transformations

## Description
Resolves: https://github.com/WordPress/gutenberg/issues/25231

This PR is the first step for handling block variation transformations. It introduces a `transform` new option for `scope` field in block variations. This way we can control how and when to handle these transformations, as this behavior doesn't always make sense for all variations for every block.

In this PR we support block variations that change only attributes, like the `Navigation` block (which is the only one adjusted for testing reasons). The design is not finalized yet and I've implemented it with a simple select control for now, just to test how this feels.

The second step in a follow up PR will be to handle transformations with `InnerBlocks` like Query or Columns that will need an API to manually handle such block variation transformations. For example what happens when we have Columns (4) and we want to expose the transform to Columns (3) etc... 

## Testing Instructions

1. Insert and init a Navigation Block (create from top level pages)
2. Select the Navigation block (not any child of it)
3. The select box is shown below the Block Card after selecting the block based on the position that seems to be prevailing at issue's discussions(https://github.com/WordPress/gutenberg/issues/25231#issuecomment-720864900)



## Checklist:
